From f0355f9add1a1224131a61b0573d806c895c21f0 Mon Sep 17 00:00:00 2001
From: Yimin Peng <yiminp@codeaurora.org>
Date: Wed, 6 Dec 2017 15:31:04 +0800
Subject: [PATCH] psci: Forbid psci reset call.

On gvm, panic process will call psci reset and then vCPU thread will
killed. To ensure pvm can receive the virtual watchdog bite event, we
must forbid the reset and stop in a deadloop there.

Change-Id: I70d33c83906e385c1ad7dd99f7b704b1c33fea94
Signed-off-by: Yimin Peng <yiminp@codeaurora.org>
---
 drivers/firmware/psci.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/firmware/psci.c b/drivers/firmware/psci.c
index c7382d4..8f82286 100644
--- a/drivers/firmware/psci.c
+++ b/drivers/firmware/psci.c
@@ -24,6 +24,9 @@
 #include <linux/reboot.h>
 #include <linux/slab.h>
 #include <linux/suspend.h>
+#ifdef CONFIG_MSM_GVM_QUIN
+#include <linux/delay.h>
+#endif
 
 #include <uapi/linux/psci.h>
 
@@ -213,6 +216,10 @@ static int get_set_conduit_method(struct device_node *np)
 
 static void psci_sys_reset(enum reboot_mode reboot_mode, const char *cmd)
 {
+#ifdef CONFIG_MSM_GVM_QUIN
+	pr_info("mdelay 20s for wdog bite!\n");
+	mdelay(20000);
+#endif
 	invoke_psci_fn(PSCI_0_2_FN_SYSTEM_RESET, 0, 0, 0);
 }
 
-- 
1.9.1

